src/crimson/os/seastore/seastore.cc

```diff
@@ -1584,6 +1584,69 @@ SeaStore::Shard::_write(
     });
 }
 
+SeaStore::Shard::tm_ret
+SeaStore::Shard::_clone_omaps(
+  internal_context_t &ctx,
+  OnodeRef &onode,
+  OnodeRef &d_onode,
+  const omap_type_t otype)
+{
+  return trans_intr::repeat([&ctx, onode, d_onode, this, otype] {
+    return seastar::do_with(
+      std::optional<std::string>(std::nullopt),
+      [&ctx, onode, d_onode, this, otype](auto &start) {
+      auto& layout = onode->get_layout();
+      return omap_list(
+	*onode,
+	otype == omap_type_t::XATTR
+	  ? layout.xattr_root
+	  : layout.omap_root,
+	*ctx.transaction,
+	start,
+	OMapManager::omap_list_config_t().with_inclusive(false, false)
```

Comment:
do we need .without_max() here?

Reply:
Not really, as we are copying attrs/omaps repeatedly here.

Reply:
why not use without_max() get all omap_list at once?  which one is efficient?

Reply:
> why not use without_max() get all omap_list one time? which one is efficient?

This may lead to large memory usage if there's a lot of attrs/omaps

Reply:
we need balance the memory and efficiency, when do repeat you need search the tree from root every time.  what is your opinion @athanatos @cyx1231st ?

Reply:
> we need balance the memory and efficiency, when do repeat you need search the tree from root every time. what is your opinion @athanatos @cyx1231st ?

The balance is done through the configuration of `omap_list_config_t::max_result_size`.

Reply:
I see, thanks.